src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+                         Sgens::Vector{<:RingElem}, R::MPolyRing) 
+
+    
+    with_v_deg_1 = [g for g in Igens if isone(degree(g,v))] 
+    length(with_v_deg_1) != 0 || return "can't isolate"
+
+    for f in with_v_deg_1
+
+        den = coefficient_v(v, f)
+        fac_den = poly_2_factors(den)
+        !issubset(fac_den, Sgens) && continue
+
+        no_v = [term(f,i) for i in 1:length(f) if !(v in vars(monomial(f,i)))]
+        iszero(length(no_v)) && continue
+              
+        h = R(-1)*sum(no_v)
+        return h//den
+        
+    end
+    return "can't solve for v"
+end
+
+
+# v is replaced by t in f
+function sub_map(v::RingElem, t::RingElem, R::MPolyRing, xs::Vector{<:RingElem}) 
+    xs_v = map(x -> x==v ? t : x, xs )    
+    return hom(R,FractionField(R), a->a, xs_v)
+end
+
+
+
+# replace v by t in f, only return the numerator.
+function sub_v(v::RingElem, t::RingElem, f::RingElem, R::Ring, xs::Vector{<:RingElem}) 
+    m = sub_map(v,t,R,xs) 
+    new_f = numerator(m(f))
+    return new_f 
+end
+
+
+# removes factors that are in the semigroup generated by Sgens
+function clean(f::RingElem, R::MPolyRing, Sgens::Vector{<:RingElem})   
+    
+    fFactors = factor(f)
+    FactorsDict = Dict(fFactors) 
+    cleanf_arr = [k^(FactorsDict[k]) for k in keys(FactorsDict) if !(k in Sgens) || is_unit(k)]
+    
+    length(cleanf_arr) > 0 ? prod(cleanf_arr) : unit(fFactors)
+    
+end
+
+#variables in ideal
+function ideal_vars(Igens::Vector{<:RingElem}) 
+    return unique!(vcat([vars(gen) for gen in Igens]...))
+end
+
+function n_new_Sgens(x::RingElem, t::RingElem, Sgens::Vector{<:RingElem}, 
+                     R::Ring, xs::Vector{<:RingElem}) 
+    preSgens = unique!([sub_v(x, t, f, R, xs) for f in Sgens])
+    return gens_2_factors(preSgens)
+end
+
+function n_new_Igens(x::RingElem, t::RingElem, Igens::Vector{<:RingElem}, 
+                     Sgens::Vector{<:RingElem}, R::Ring, xs::Vector{<:RingElem}) 
+
+    preIgens = unique!([clean(sub_v(x, t, f, R, xs), R, Sgens) for f in Igens])
+    return filter(x-> x!= R(0), preIgens)
+end
+
+
+
+function matrix_clear_den_in_col(X::Oscar.MatElem, c::Int)
+    Xc = [denominator(f) for f in X[:, c]]
+    t = lcm(Xc)
+    result = multiply_column!(X, t, c)
+    return result
+
+end
+
+
+
+function matrix_clear_den(X::Oscar.MatElem)
+    rs, cs = size(X)
+    for c in 1:cs
+        X = matrix_clear_den_in_col(X, c)
+    end
+    return X
+end
+
+
+function reduce_ideal_one_step(MRS::MatroidRealizationSpace, 
+                               elim::Vector{<:RingElem}, 
+                               fullyReduced::Bool)
+
+    Igens = gens(MRS.defining_ideal)
+    Sgens = MRS.inequations
+    R = MRS.ambient_ring
+    FR = FractionField(R)
+    xs = gens(R)
+    X = MRS.realization_matrix
+    nr, nc = size(X)
+    
+    Ivars = ideal_vars(Igens);
+
+    for x in Ivars 
+        t = find_solution_v(x, Igens, Sgens, R)
+        t isa String && continue
```

Comment:
If you want to signal "no solution found", then return `nothing` or `missing`, but not an error string.

Reply:
Changed this to a try/catch. I think it should work. The idea is that if no solution is found, then it just goes to the next variable. If none of the variables can be solved for, then the recursion ends.